code/game/machinery/suit_storage_unit.dm

```diff
@@ -1,184 +1,406 @@
+/obj/machinery/suit_storage_unit/proc/resist_open(mob/user)
+	if(!state_open && occupant && (user in src) && user.stat == 0) // Check they're still here.
+		visible_message("<span class='notice'>You see [user] burst out of [src]!</span>", \
+			"<span class='notice'>You escape the cramped confines of [src]!</span>")
+		open_machine()
+
+//eventually move these onto the parent....
+
+/obj/machinery/suit_storage_unit/proc/open_machine(drop = TRUE)
+	state_open = TRUE
+	density = FALSE
+	if(drop)
+		dropContents()
+	update_icon()
+	updateUsrDialog()
+
+/obj/machinery/suit_storage_unit/dropContents()
+	var/turf/T = get_turf(src)
+	for(var/atom/movable/A in contents)
+		A.forceMove(T)
+		if(isliving(A))
+			var/mob/living/L = A
+			L.update_canmove()
+	occupant = null
+
+/obj/machinery/suit_storage_unit/proc/close_machine(atom/movable/target = null)
+	state_open = FALSE
+	density = TRUE
+	if(!target)
+		for(var/am in loc)
+			if(!(occupant_typecache ? is_type_in_typecache(am, occupant_typecache) : isliving(am)))
+				continue
+			var/atom/movable/AM = am
+			if(AM.has_buckled_mobs())
+				continue
+			if(isliving(AM))
+				var/mob/living/L = am
+				if(L.buckled || L.mob_size >= MOB_SIZE_LARGE)
+					continue
+			target = am
+
+	var/mob/living/mobtarget = target
+	if(target && !target.has_buckled_mobs() && (!isliving(target) || !mobtarget.buckled))
+		occupant = target
+		target.forceMove(src)
+	updateUsrDialog()
+	update_icon()
 
 
+////////
+
 /obj/machinery/suit_storage_unit/attack_hand(mob/user as mob)
 	var/dat
 	if(..())
 		return
 	if(stat & NOPOWER)
 		return
 	if(!user.IsAdvancedToolUser())
-		return 0
-	if(src.panelopen) //The maintenance panel is open. Time for some shady stuff
+		return FALSE
+	if(panel_open) //The maintenance panel is open. Time for some shady stuff
 		dat+= "<HEAD><TITLE>Suit storage unit: Maintenance panel</TITLE></HEAD>"
 		dat+= "<B>Maintenance panel controls</B><HR>"
 		dat+= "<font color ='grey'>The panel is ridden with controls, button and meters, labeled in strange signs and symbols that <BR>you cannot understand. Probably the manufactoring world's language.<BR> Among other things, a few controls catch your eye.<BR><BR>"
-		dat+= text("A small dial with a \"�\" symbol embroidded on it. It's pointing towards a gauge that reads [].<BR> <font color='blue'><A href='?src=[UID()];toggleUV=1'> Turn towards []</A><BR>",(src.issuperUV ? "15nm" : "185nm"),(src.issuperUV ? "185nm" : "15nm") )
-		dat+= text("A thick old-style button, with 2 grimy LED lights next to it. The [] LED is on.<BR><font color ='blue'><A href='?src=[UID()];togglesafeties=1'>Press button</a></font>",(src.safetieson? "<font color='green'><B>GREEN</B></font>" : "<font color='red'><B>RED</B></font>"))
+		dat+= text("A small dial with a \"�\" symbol embroidded on it. It's pointing towards a gauge that reads [].<BR> <font color='blue'><A href='?src=[UID()];toggleUV=1'> Turn towards []</A><BR>",(uv_super ? "15nm" : "185nm"),(uv_super ? "185nm" : "15nm") )

```

Comment:
what's this � symbol?

also `It's pointing towards a gauge that reads []`

Reply:
Explanation would be welcome